client/server/user-bootstrap/test/index.js

```diff
@@ -0,0 +1,361 @@
+			},
+		} );
+		const request = mockRequest( {
+			cookies: {
+				wordpress_logged_in: 'auth-cookie',
+			},
+		} );
+
+		await getBootstrappedUser( request );
+
+		expect( upstreamRequest.isDone() ).toBe( true );
+	} );
+
+	it( 'sets the support session cookie', async () => {
+		const upstreamRequest = configureUpstreamRequest( {
+			headers: {
+				cookie: ( val ) => val.includes( 'support_session_id=session-id' ),
+			},
+		} );
+		const request = mockRequest( {
+			cookies: {
+				wordpress_logged_in: 'auth-cookie',
+				support_session_id: 'session-id',
+			},
+		} );
+
+		await getBootstrappedUser( request );
+
+		expect( upstreamRequest.isDone() ).toBe( true );
+	} );
+
+	describe( 'with support session header', () => {
+		let request;
+		let upstreamRequest;
+
+		beforeEach( () => {
+			request = mockRequest( {
+				cookies: {
+					wordpress_logged_in: 'auth-cookie',
+				},
+				headers: {
+					'x-support-session': 'session-id',
+				},
+			} );
+			upstreamRequest = configureUpstreamRequest( {
+				headers: {
+					// Hardcoded value resulting from hashing "key" + "session-id"
+					Authorization: 'X-WPCALYPSO-SUPPORT-SESSION bc8844a622734cad18a2ec733e11b296',
+					'x-support-session': 'session-id',
+				},
+			} );
+		} );
+
+		it( 'throws if there is no support session API key', async () => {
+			withConfig( {
+				wpcom_calypso_support_session_rest_api_key: undefined,
+			} );
+
+			await expect( getBootstrappedUser( request ) ).rejects.toThrow(
+				new Error(
+					'Unable to boostrap user because of invalid SUPPORT SESSION API key in secrets.json'
+				)
+			);
+		} );
+
+		it( 'sets the authorization header', async () => {
+			withConfig( {
+				wpcom_calypso_support_session_rest_api_key: 'key',
+			} );
+
+			await getBootstrappedUser( request );
+
+			expect( upstreamRequest.isDone() ).toBe( true );
+		} );
+
+		it( 'reads support session API key from env variables', async () => {
+			withConfig( {
+				wpcom_calypso_support_session_rest_api_key: undefined,
+			} );
+			withEnv( {
+				WPCOM_CALYPSO_SUPPORT_SESSION_REST_API_KEY: 'key',
+			} );
+
+			await getBootstrappedUser( request );
+
+			expect( upstreamRequest.isDone() ).toBe( true );
+		} );
+
+		it( 'support session API key from env variable overrides config', async () => {
+			withConfig( {
+				wpcom_calypso_support_session_rest_api_key: 'old_key',
+			} );
+			withEnv( {
+				WPCOM_CALYPSO_SUPPORT_SESSION_REST_API_KEY: 'key',
+			} );
+
+			await getBootstrappedUser( request );
+
+			expect( upstreamRequest.isDone() ).toBe( true );
+		} );
+	} );
+
+	describe( 'with auth session', () => {
+		let request;
+		let upstreamRequest;
+
+		beforeEach( () => {
+			request = mockRequest( {
+				cookies: {
+					wordpress_logged_in: 'auth-cookie',
+				},
+			} );
+			upstreamRequest = configureUpstreamRequest( {
+				headers: {
+					// Hardcoded value resulting from hashing "key" + "auth-cookie"
+					Authorization: 'X-WPCALYPSO 26be6ad9e36fde3770b1e81a559db109',
```

Comment:
This should be expecting also the `wordpress_logged_in` cookie, just like the mock request for support sessions expects both `X-WPCALYPSO-SUPPORT-SESSION` auth header and the `x-support-session` header.

In both cases, they should always travel together in pairs. The auth cookie or support session token, plus the `X-WPCALYPSO` HMAC that confirms that the requester knows not only the auth credentials that travel over network, but also the secret keys.

Reply:
Added